### PR TITLE
Adds FakeGoogleCredentials for unit tests to stop relying on user bei…

### PR DESCRIPTION
**Pull request by ttomsu**
…ng gcloud auth'ed

Tested with:
```
$ gcloud auth revoke
$ ./gradlew clean test
```

@duftler PTAL
@cfieber - I don't know how to properly add a dependency on another modules _test_ classes. Does this approach look right to you?

### CODE REVIEW

File: clouddriver-google/src/test/groovy/com/netflix/spinnaker/clouddriver/google/security/FakeGoogleCredentials.groovy [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    this("unitTestProject")
+  }
+
+  FakeGoogleCredentials(String project) {
+    super(project, ComputeVersion.V1)
+  }
+
+  @Override
+  HttpTransport buildHttpTransport() {
+    return new MockHttpTransport.Builder().build()
```
Comment (duftler) [inline clouddriver-google/src/test/groovy/com/netflix/spinnaker/clouddriver/google/security/FakeGoogleCredentials.groovy]: Nice.

**Comment by cfieber** [comment]
Just taking a quick look at an alternative to referencing that test class

**Comment by duftler** [comment]
LGTM once @cfieber is happy with the test dependency handling.

**Comment by cfieber** [comment]
have a look at https://github.com/ttomsu/clouddriver/pull/1 - it makes sure the test runtime dependencies come along for the ride in addition to the classes, and it is supposed to play nicer with IDEs.

otherwise LGTM, thanks for the fix!

**Pull request merged**